services/app/src/actions/Settings.tsx

```diff
@@ -1,7 +1,35 @@
+import {MultiplayerState, SettingsType} from '../reducers/StateTypes';
 import {ChangeSettingsAction} from './ActionTypes';
 
 export function changeSettings(settings: any): ChangeSettingsAction {
   // TODO: Changing player count in settings should affect
   // player status
   return {type: 'CHANGE_SETTINGS', settings};
 }
+
+export function numAdventurers(settings: SettingsType, rp: MultiplayerState): number {
+  if (!rp || !rp.clientStatus || Object.keys(rp.clientStatus).length < 2) {
+    // Since single player still has two adventurers, the minimum possible is two.
+    return Math.max(2, settings.numLocalPlayers);
+  }
+  return countnumLocalPlayers(rp);
+}
+
+export function numLocalPlayers(settings: SettingsType, rp?: MultiplayerState): number {
```

Comment:
I think this should stay `numPlayers`, since it's counting players from non-local devices.